### DISCUSSION TITLE
Bundler: Ability To Comment Bundle Lists Should Be A Default?

### DISCUSSION THREAD

**Discussion by TiddlyTweeter** [wikilabs/plugins#60] [Ideas]
@pmario, I am a fan of the Bundler. With it you can maintain an entire ecosystem for QUICKLY configuring wikis. I been using it a long time successfully with no problems.

But, for best usage, I have to HACK $:/plugins/wikilabs/bundler/ui/BundlesList ... so a \define will allow me to COMMENT in the bundles ...

> \define filter-exist() [enlist{$(currentTiddler)$}] +[has[title]]
> _**\define filter-list() [enlist{$(currentTiddler)$}!prefix[--- ]]**_

My QUESTION ... why don't you make this ability to add comments as the default? If you don't then counts of list bundle contents don't add up. Both Comments & accurate Enumeration matter if you are routinely using the bundler to maintain wikis, I think.

OR alternatively a Comment Construct starting "\\\\" so it visually matches your recent [https://wikilabs.github.io/editions/pragma-comment/](https://wikilabs.github.io/editions/pragma-comment/) plugin? (would not need it present, merely **visually** echo it).

Thoughts on making a very, very good tool slightly better :-)

Best wishes
TT

**Comment by pmario** [comment]
@TiddlyTweeter .. Do I understand it right, that you want `prefix[---]` to be the indicator for comment-tiddlers in bundler?

eg:

```
[[--- !This tiddler ---]]
$:/tt/bun/search-en-dictionaries

[[--- Search Online English Dictionaries ---]]
```

**Reply by pmario** [reply]
I'm thinking about it

**Reply by TiddlyTweeter** [reply]
> @TiddlyTweeter .. Do I understand it right, that you want `prefix[---]` to be the indicator for comment-tiddlers in bundler?
> 
> eg:
> 
> ```
> [[--- !This tiddler ---]]
> $:/tt/bun/search-en-dictionaries
> 
> [[--- Search Online English Dictionaries ---]]
> ```

@pmario, actually I been using "**prefix[--- ]**" since 2018 when [we last discussed this](https://groups.google.com/g/tiddlywiki/c/ZPaiDtpt1Zc/m/AUZBac0PBQAJ) :-) and you posted ...

> To test the theory, just open the tiddler:  $:/plugins/wikilabs/bundler/ui/Bundles and change the 2nd line
> 
> from:
> 
> \define filter-list() [enlist{$(currentTiddler)$}]
> 
> to:  
> 
> \define filter-list() [enlist{$(currentTiddler)$}!prefix[--- ]] 
> 
> So the new convention would be, that the title-prefix for a "comment-tiddler" is 3 dashes + a space: "--- "

**Reply by TiddlyTweeter** [reply]
> @TiddlyTweeter .. Do I understand it right, that you want `prefix[---]` to be the indicator for comment-tiddlers in bundler?

@pmario but further to my last note, though I have been using **prefix[--- ]** since 2018, I DO NOT MIND what the comment prefix is, just that its an agreed pattern that all Bundler users could use if they wanted.

In my first note I suggested that lines starting "**\\\\**" might work? I like that because its the same _visual_ pattern as your recent pragma comment plugin.

**Reply by pmario** [reply]
Bundler plugin was developed short after the `enlist` filter operator was introduced. It only contains "vanilla" TW functions, without the need to modify filter operators. 

We only modify the import behaviour. 

Introducing a "special" comment marker like `\\` would need to modify the enlist filter. I'm sure Jeremy wouldn't merge that. The PR "add pragmacomment wikirule #5287" hasn't been merged, because there is no consensus about the marker yet ...

**Reply by pmario** [reply]
I'll make `prefix[--- ]]` ... tripple-dash + space ... the "standard". It's a "convention", which seems to work well and we didn't get any responses for 2+ years. .. So it should be OK:

**Reply by TiddlyTweeter** [reply]
> I'll make `prefix[--- ]]` ... tripple-dash + space ... the "standard". It's a "convention", which seems to work well and we didn't get any responses for 2+ years. .. So it should be OK:

Great from my point of view! Thanks!

**Comment by pmario** [comment]
The changelog atm looks like this: 

```
* Make `prefix[--- ]]` the standard for tiddler titles, that are used as comments
** It's easy to read and we didn't get complains for 2 years, since the suggestion was made

* Bundler import allows us to import `$:/state/` tiddlers.
** The plugin overwrites [[$:/core/modules/upgraders/system.js]] now

* $:/StoryList and $:/HistoryList are prefixes now. 
** `DONT_IMPORT_PREFIX_LIST = ["$:/temp/","$:/Import","$:/StoryList","$:/HistoryList"],`
** This will allow us to work with multiple stories
```

I'm interested, what you think about the possibility to import `$:/state/` tiddlers?

**Reply by TiddlyTweeter** [reply]
> @pmario I'm interested, what you think about the possibility to import `$:/state/` tiddlers?

I think it is a VERY good idea!

On import you can always de-select what you don't want. But the fact is in some use cases I have is that  `$:/state/` tiddlers are really working as "config" items. Item that are meant to persist. Being able to bundle and successfully import them would be great from my point of view.
